### DISCUSSION TITLE
No matching version found for @eclipse-glsp/layout-elk@0.9.0

### DISCUSSION THREAD

**Discussion by gee4vee** [eclipse-glsp/glsp#589] [Q&A] [answered]
I am trying to run `npm install` from within the `glsp-server-node/examples/workflow-server` directory but I get this error:

```
~/g/e/g/e/workflow-server [main]{0} ⛅» npm install                                                                                                                       {15:13:18}
npm ERR! code ETARGET
npm ERR! notarget No matching version found for @eclipse-glsp/layout-elk@0.9.0.
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.
```

Is this the correct approach? Should I be running from the root project directory?

This error also leads me to wonder how one would create their own server. Do we just have something like what's within the `examples/workflow-server` directory or do we need to also include the contents of the `glsp-server-node` directory?

**Comment by tortmayr** [comment, marked as answer]
Hi @gee4vee ,
> Is this the correct approach? Should I be running from the root project directory?

We are using `yarn` as package manager. So  `npm install` probably wont work.
You can build all packages by executing `yarn` in the package root. 
If you just want to build the example server you can execute `yarn build` from within `glsp-server-node/examples/workflow-server`.

>This error also leads me to wonder how one would create their own server. Do we just have something like what's within the examples/workflow-server directory or do we need to also include the contents of the glsp-server-node directory?

If you want to create your own server you would have to implement a package similar to the  `examples/workflow-server`. You don't need the contents of the `glsp-server-node` directory. In your own example these packages can be consumed via [`npm`](https://www.npmjs.com/search?q=%40eclipse-glsp%2Fserver-node).

We have started working on the node based glsp-server pretty recently which means there is no official release yet. If you wan to create your own server you can to use the `next` versions from npm for now. So you would have to change the dependencies in the `package.json`  of you server package:

```ts
// from
{
...
  "dependencies": {
    "@eclipse-glsp/server-node": "0.9.0",
    "@eclipse-glsp/layout-elk": "0.9.0"
  },
}

// to
{
...
  "dependencies": {
    "@eclipse-glsp/server-node": "next", // or a concrete next version e.g. `0.9.0-next.95961ba.7`
    "@eclipse-glsp/layout-elk": "next" 
  },
}
```

**Reply by gee4vee** [reply]
Thanks!

**Reply by gee4vee** [reply]
Another question @tortmayr ... after doing `yarn build` within `examples/workflow-server`, I see errors throughout the code about experimental decorator support and things like `Accessors are only available when targeting ECMAScript 5 and higher`. I set `experimentalDecorators` to `true` in `tsconfig.json` and rebuilt but no change. What should we do to get VS Code errors like these to resolve?

**Reply by tortmayr** [reply]
Could you please post a code snipped of your `ts.config`? This makes it easier for us to identify potential issues.

**Reply by gee4vee** [reply]
I made a copy of the workflow-server project as a starting point and most of those went away. However still seeing this error in tsconfig.json:

```
Cannot find type definition file for 'mocha'.
  The file is in the program because:
    Entry point of type library 'mocha' specified in compilerOptions
```

Here is my file:
```
{
  "extends": "@eclipse-glsp/ts-config/tsconfig.json",
  "compilerOptions": {
    "rootDir": "src",
    "outDir": "lib",
    "reactNamespace": "JSX",
    "baseUrl": ".",
    "types": ["node", "mocha", "reflect-metadata"],
    "experimentalDecorators": true
  },
  "include": ["src"],
  "exclude": ["**/*.spec.ts"]
}
```

**Reply by gee4vee** [reply]
Correction: still seeing the error `Accessors are only available when targeting ECMAScript 5 and higher.` in a couple places with above tsconfig.json.
